web/client/utils/MapInfoUtils.js

```diff
@@ -91,6 +159,37 @@ export const getLayerFeatureInfoViewer = (layer) => {
 export const getLayerFeatureInfo = (layer) => {
     return layer && layer.featureInfo && {...layer.featureInfo} || {};
 };
+/**
+ * @return {string} the info format value from layer, otherwise the info format in settings
+ */
+
+export const getDefaultInfoFormatValueFromLayer = (layer, props) => {
+    const featInfoFormat = getLayerFeatureInfo(layer)?.format;
+    if (featInfoFormat) {
+        // When the user explicitly configures the format from the layer settings => feature info page, return directly from definition map.
+        return getInfoFormatByInfoView(featInfoFormat, layer.infoFormats) || INFO_FORMATS;
+    }
+    if (props.format) {
+        if (props.format === JSON_MIME_TYPE && layer.infoFormats && layer.infoFormats.includes(GEOJSON_MIME_TYPE)) {
+            // When global settings is configured for PROPERTIES (json), layer settings are not used and the layer.info_format configuration supports geo+json
+            // then override global settings and set param.info_format to geo+json mime type explicitly.
+            return GEOJSON_MIME_TYPE;
+        }
+
+        // otherwise, preserve and obey the global configration for getFeatureInfo mime type.
+        return props.format;
+    }
+
+    // if global configration somehow fails provide a last fallback.
+    return MapInfoUtils.getDefaultInfoFormatValue();
+};
+export const getLayerFeatureInfoViewer = (layer) => {
```

Comment:
document this
```suggestion
/*
 * @param {object} layer a layer object
 * @returns {object} the viewer configured for the layer. If viewer is not configured, it returns an empty object.
 */
export const getLayerFeatureInfoViewer = (layer) => {
```

Reply:
Fixed in https://github.com/geosolutions-it/MapStore2/pull/9917/commits/9b24f0df5edc2fe5a789f504b9c283475838a901